File: 001.c
// C bank application API

#include <stdbool.h>
#include <stdio.h>

double accountBalance = 1000;

double getAccountBalance() {
  return accountBalance;
}

void deposit(double amount) {
  if (amount <= 0) {
    fprintf(stderr, "Invalid amount. Amount must be greater than zero.\n");
    return;
  }
  accountBalance += amount;
}

bool withdraw(double amount) {
  if (amount <= 0) {
    fprintf(stderr, "Invalid amount. Amount must be greater than zero.\n");
    return false;
  }
  if (amount > accountBalance) {
    fprintf(stderr, "Insufficient balance.\n");
    return false;
  }
  accountBalance -= amount;
  return true;
}

File: 001.java
// Java bank application API

public class BankApplicationAPI {
    private static double accountBalance = 1000;

    public static double getAccountBalance() {
        return accountBalance;
    }

    public static void deposit(double amount) {
        if (amount <= 0) {
            throw new IllegalArgumentException("Invalid amount. Amount must be greater than zero.");
        }
        accountBalance += amount;
    }

    public static boolean withdraw(double amount) {
        if (amount <= 0) {
            throw new IllegalArgumentException("Invalid amount. Amount must be greater than zero.");
        }
        if (amount > accountBalance) {
            throw new IllegalArgumentException("Insufficient balance.");
        }
        accountBalance -= amount;
        return true;
    }
}

